src/transformers/models/fastvit/modeling_fastvit.py

```diff
@@ -0,0 +1,1103 @@
+        embeddings = self.patch_embeddings(pixel_values)
+        embeddings = self.dropout(embeddings)
+
+        return embeddings
+
+
+class FastViTPatchEmbeddings(nn.Module):
+    """
+    Construction of the Stem Block, following paper structure here <https://arxiv.org/abs/2303.14189>.
+    """
+
+    def __init__(self, config: FastViTConfig) -> None:
+        super().__init__()
+        image_size = config.image_size
+        num_channels, hidden_size = config.num_channels, config.hidden_sizes[0]
+
+        image_size = image_size if isinstance(image_size, collections.abc.Iterable) else (image_size, image_size)
+        self.image_size = image_size
+        self.num_channels = num_channels
+        self.inference = config.inference
+        self.config = config
+
+        self.projection = nn.Sequential(
+            FastViTConvLayer(
+                in_channels=num_channels,
+                out_channels=hidden_size,
+                kernel_size=3,
+                stride=2,
+                padding=1,
+                groups=1,
+                inference=self.inference,
+            ),
+            FastViTConvLayer(
+                in_channels=hidden_size,
+                out_channels=hidden_size,
+                kernel_size=3,
+                stride=2,
+                padding=1,
+                groups=hidden_size,
+                inference=self.inference,
+            ),
+            FastViTConvLayer(
+                in_channels=hidden_size,
+                out_channels=hidden_size,
+                kernel_size=1,
+                stride=1,
+                padding=0,
+                groups=1,
+                inference=self.inference,
+            ),
+        )
+
+    def forward(self, pixel_values: torch.Tensor) -> torch.Tensor:
+        batch_size, num_channels, height, width = pixel_values.shape
+
+        if num_channels != self.num_channels:
+            raise ValueError(
+                "Make sure that the channel dimension of the pixel values match with the one set in the configuration."
+                f" Expected {self.num_channels} but got {num_channels}."
+            )
+
+        embeddings = self.projection(pixel_values)
+
+        return embeddings
+
+
+class FastViTConvLayer(nn.Module):
+    """
+    Build of Convolution Layer following structure proposed in <https://arxiv.org/pdf/2303.14189.pdf> This block has a
+    multi-branched architecture at train-time and plain-CNN style architecture at inference time
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int = 1,
+        padding: int = 0,
+        dilation: int = 1,
+        groups: int = 1,
+        inference: bool = False,
+        num_conv_branches: int = 1,
+        use_scale_branch: bool = True,
+        use_act: bool = True,
+        use_se: bool = False,
+        activation: nn.Module = nn.GELU(),
```

Comment:
Some if not most of these should come from the config (of course not :
```python 
                in_channels=hidden_size,
                out_channels=hidden_size,
                kernel_size=1,
                stride=1,
                padding=0,
                groups=1,
```
inference should also IMO be just if self.training() no?